dbt_style_guide.md

```diff
@@ -1,237 +1,415 @@
+  Example:
+  ```sql
+  transformed as (
+
+      select
+
+        -- ids
+        order_id,
+        customer_id,
+
+        -- dimensions
+        order_status,
+        is_shipped,
+
+        -- measures
+        order_total,
+
+        -- metdata
+        created_at,
+        updated_at,
+        _sdc_batched_at
+
+      from source
+
+  )
+  ```
+
+- Date/time columns should be named according to these conventions:
+  - Timestamps: `<event>_at`
+    Format: UTC  
+    Example: `created_at`
+  
+  - Dates: `<event>_date`  
+    Format: Date  
+    Example: `created_date`
+
+- Booleans should be prefixed with `is_` or `has_`.
+
+- Price/revenue fields should be in decimal currency (e.g. `19.99` for $19.99; many app databases store prices as integers in cents). If non-decimal currency is used, indicate this with suffix, e.g. `price_in_cents`.
+
+- Avoid using reserved words (such as [these](https://docs.snowflake.com/en/sql-reference/reserved-keywords.html) for Snowflake) as column names.
+
+- Consistency is key! Use the same field names across models where possible.  
+Example: a key to the `customers` table should be named `customer_id` rather than `user_id`.
 
 ## CTEs
 
-For more information about why we use so many CTEs, check out [this discourse post](https://discourse.getdbt.com/t/why-the-fishtown-sql-style-guide-uses-so-many-ctes/1091).
+For more information about why we use so many CTEs, check out [this glossary entry](https://docs.getdbt.com/terms/cte).
 
-- All `{{ ref('...') }}` statements should be placed in CTEs at the top of the file
 - Where performance permits, CTEs should perform a single, logical unit of work.
-- CTE names should be as verbose as needed to convey what they do
-- CTEs with confusing or noteable logic should be commented
-- CTEs that are duplicated across models should be pulled out into their own models
-- create a `final` or similar CTE that you select from as your last line of code. This makes it easier to debug code within a model (without having to comment out code!)
-- CTEs should be formatted like this:
 
-``` sql
-with
+- CTE names should be as verbose as needed to convey what they do.
 
-events as (
+- CTEs with confusing or noteable logic should be commented with SQL comments, as you would with any complex functions.
 
-    ...
+- CTEs that are duplicated across models should be pulled out and created as their own models.
 
-),
+- CTEs and SQL within a model should follow this structure:
+  - `with` statement
+  - Import CTEs
+  - Logical CTEs
+  - Final CTE
+  - Simple `select` statement
 
--- CTE comments go here
-filtered_events as (
+- All `{{ ref('...') }}` statements should be placed in CTEs at the top of the file ("import" CTEs)
 
-    ...
+- SQL should end with a simple select statement. All other logic should be contained within CTEs.  
+  Example: `select * from final`
 
-)
+- Where applicable, opt for filtering within import CTEs over filtering within logical CTEs.

```

Comment:
Do we define what a logical cte is? I see we name `"import" CTEs` - can we do the same for final and logical?

Reply:
@graciegoheen Do you have a good example of what you're thinking here?

Reply:
I believe this just means if you are joining some tables and always filtering something that could be moved up to the top, like:
```
with
customers as (
  select * from {{ ref('stg_customers') }} where is_deleted is false
)
```

Reply:
@graciegoheen Can you see if [this new structure](https://github.com/dbt-labs/corp/pull/63/commits/ad31e8a610c7ffb17cbf2bff901c7794f5c725d5) helps?